### ISSUE TITLE
A11y: CheckboxGroup and LayerControlGroup are missing attributes for screen readers

### ISSUE THREAD

**Issue by halimahLDN** [Greater-London-Authority/ldn-viz-tools#935]
**Is your feature request related to a problem? Please describe.**
`CheckboxGroup` and `LayerControlGroup` components aren't being dictated by screen readers effectively, as group level labels aren't being dictated and the inputs aren't clearly connected with semantic html.

**Describe the solution you'd like**
As per the [WAI ARIA Checkbox pattern](https://www.w3.org/WAI/ARIA/apg/patterns/checkbox/examples/checkbox/):
- Each input should be inside an `<li>` element, as part of an unordered list
- The `<ul>` containing all inputs should be contained by a `<div>` with `role="group"`
- This group div should have an `aria-labelledby` attribute connecting the group to the group label

**Describe alternatives you've considered**
N/A

**Additional context**
